src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -2445,7 +2445,7 @@ private static void ReportReadOnlyError(Symbol symbol, SyntaxNode node, BindValu
             };
 
             int index = (checkingReceiver ? 3 : 0) + (kind == BindValueKind.RefReturn ? 0 : (RequiresRefOrOut(kind) ? 1 : 2));
-            Error(diagnostics, ReadOnlyErrors[index], node, symbolKind, symbol);
+            Error(diagnostics, ReadOnlyErrors[index], node, symbolKind, new FormattedSymbol(symbol, SymbolDisplayFormat.ShortFormat));
```

Comment:
>new FormattedSymbol(symbol, SymbolDisplayFormat.ShortFormat) [](http://example.com/codeflow?start=72&length=60)

I think we should investigate an alternative of changing default behavior for ```CSharpErrorMessageFormat``` and ```CSharpShortErrorMessageFormat```. It feels more appropriate to include parameter names by default when displaying parameter symbols. There is a comment there: "Not showing the name is important because we visit parameters to display their types.  If we visited their types directly, we wouldn't get ref/out/params." It feels like consumers that are interested only in types of parameters should use a special display format, that we can expose publicly as well.

Reply:
If it makes sense for a diagnostics to include parameter name, the baseline should be updated. If it doesn't make sense, the code reporting it should be updated. 

> this would require lots of changes (I think)

This is quite possible, but I think it is worth to explore this option.

Reply:
It feels like it might be worth taking this change as is for now, so it can get in for 7.0/17.4, and filing a follow-up issue to look at the bigger refactoring for 17.5? I agree the larger change would be good (I've thought about it before myself), but the scope feels like it could balloon quickly and I'd prefer to give broad SymbolDisplay changes more bake time in VS previews, rather than going out with very little preview for 17.4.

Reply:
> It feels like it might be worth taking this change as is for now, so it can get in for 7.0/17.4,

I do not think I agree with that. And I don't see the urgency to go with a fix for 17.4, the problem was there for years.

> I'd prefer to give broad SymbolDisplay changes more bake time in VS previews, rather than going out with very little preview for 17.4.

I don't think there was a suggestion to change behavior of ```SymbolDisplay```. The suggestion was to change default formatting for parameters in error messages.

Reply:
> The suggestion was to change default formatting for parameters in error messages.

These are public properties, and changing the way we display in error messages seems extremely likely to me to affect the analyzer ecosystem. There are at least 30 non-roslyn references to `CSharpErrorMessageFormat`. This is what I referring to.

> And I don't see the urgency to go with a fix for 17.4, the problem was there for years.

Generally, I don't see the issue with improving some diagnostics, and pursuing a more general improvement as a followup step.

Reply:
> These are public properties, and changing the way we display in error messages seems extremely likely to me to affect the analyzer ecosystem. There are at least 30 non-roslyn references to CSharpErrorMessageFormat. This is what I referring to.

I think chances of introducing a real problem are extremely low. This is based on the following assumptions that I think are reasonable to make:
- Whoever is using the format is delegating to us the decision to use an appropriate formatting. 
- A reasonable person is going to expect that parameter name is included into an error message.

Also, we did not make an explicit commitment to format symbols in any specific way when that option is used.

Reply:
I don't think it's an unreasonable change. I'd just prefer to make it as part of a preview 1, not a preview 3/4.

Reply:
I guess there is no harm in doing this in two steps. How do you want to trach that?

Reply:
I think not closing the original issue and updating it to track the remaining work would be fine.

Reply:
Added [a comment](https://github.com/dotnet/roslyn/issues/62096#issuecomment-1257594632) to the issue and changed description of this PR to avoid closing that issue.